compliance/index/v4/convert_test.go

```diff
@@ -0,0 +1,309 @@
+package v4
+
+import (
+	"testing"
+
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/protoassert"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestIndexReportConvert(t *testing.T) {
+	suite.Run(t, new(indexReportConvertSuite))
+}
+
+type indexReportConvertSuite struct {
+	suite.Suite
+}
+
+func (s *indexReportConvertSuite) TestToNodeInventory() {
+	r := createVulnerabilityReport()
+
+	actual := ToNodeScan(r)
+
+	s.Equal(storage.NodeScan_SCANNER_V4, actual.GetScannerVersion())
+	s.Equal(1, len(actual.GetComponents()))
+	s.Equal("openssh-clients", actual.GetComponents()[0].GetName())
+	s.Equal("8.7p1-38.el9", actual.GetComponents()[0].GetVersion())
+	s.Equal("RHSA-2024:4616", actual.GetComponents()[0].GetVulns()[0].GetCve())
+	s.Equal([]storage.NodeScan_Note{storage.NodeScan_UNSUPPORTED}, actual.GetNotes())
+}
+
+func (s *indexReportConvertSuite) TestEmptyReportConversionNoPanic() {
+	r := &v4.VulnerabilityReport{
+		HashId:                 "",
+		Vulnerabilities:        nil,
+		PackageVulnerabilities: nil,
+		Contents:               nil,
+		Notes:                  nil,
+	}
+
+	var actual *storage.NodeScan
+
+	s.NotPanics(func() {
+		actual = ToNodeScan(r)
+	})
+
+	s.NotNil(actual)
+	s.Equal(storage.NodeScan_SCANNER_V4, actual.GetScannerVersion())
+
+}
+
+func (s *indexReportConvertSuite) TestToStorageComponentsOutOfBounds() {
+	in := createOutOfBoundsReport()
+	expectedCVE := &storage.EmbeddedVulnerability{
+		Cve:               "RHSA-2024:4616",
+		Summary:           "Sample Description",
+		SetFixedBy:        &storage.EmbeddedVulnerability_FixedBy{FixedBy: "0:4.16.0-202407111006.p0.gfa84651.assembly.stream.el9"},
+		VulnerabilityType: storage.EmbeddedVulnerability_NODE_VULNERABILITY,
+		Severity:          storage.VulnerabilitySeverity_MODERATE_VULNERABILITY_SEVERITY,
+	}
+
+	actual := toStorageComponents(in)
+
+	s.Len(len(actual), 2)
```

Comment:
This won't work, as it is used a bit differently
```suggestion
	s.Len(actual, 2)
```